docs/mcp.md

```
@@ -0,0 +1,1194 @@
+            },
+            ToolsToExecute: []string{"read_file", "list_files"}, // Whitelist safe tools
+        },
+    },
+}
+```
+
+#### 2. **Critical Message Sequencing**
+
+The LLM expects this exact conversation flow:
+
+```
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified
null

(MD040, fenced-code-language)